src/streams/receive.rs

```diff
@@ -243,7 +247,20 @@ impl StreamRx {
         }
     }
 
-    fn set_sender_info(&mut self, now: Instant, info: SenderInfo) {
+    fn set_sender_info(&mut self, now: Instant, mut info: SenderInfo) {
+        // Extend the incoming time given our knowledge of last time.
+        let extended = {
+            let prev = self.sender_info.map(|(_, sr)| sr.rtp_time.numer() as u64);
+            let r_u32 = info.rtp_time.numer() as u32;
+            extend_u32(prev, r_u32)
+        };
+
+        // The MediaTime has a base 1 after being parsed. At this point
+        // we know whether it's audio or video and set the base accordingly.
+        let clock_base = if self.kind.is_audio() { 48_000 } else { 90_000 };
```

Comment:
Is this future-proof? I have a vague recollection that all the Pts negotiated for a given mid need to have the same clock rate

Reply:
Well. The day we have some other clock base, we deal with it.

Reply:
Sure, the only issue is we might not remember that we need to update it in this specific place